casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -14,26 +14,27 @@ import scala.concurrent.SyncVar
 import scala.util.{Failure, Success, Try}
 
 //runtime is a SyncVar for thread-safety, as all checkpoints share the same "hot store"
-class Checkpoint private (val hash: ByteString, runtime: SyncVar[Runtime]) {
+class RuntimeManager private (runtime: SyncVar[Runtime]) {
+  type StateHash = ByteString
 
-  def updated(terms: List[Par])(implicit scheduler: Scheduler): Either[Throwable, Checkpoint] = {
-    val active = getActive()
+  def updated(hash: StateHash, terms: List[Par])(
```

Comment:
I think I would prefer if this method were called `computeStateFrom` since we're not returning a new `RuntimeManager`